### ISSUE TITLE
Add docs for soak::Columns

### ISSUE THREAD

**Issue by pickfire** [driveyard/driveyard#5]
Currently there doesn't seemed to be any docs for the `soak::Columns` derive macros. Maybe we can show an example how it is used, there are no attributes right now but may be useful later.

By the way, why the name "dioptre" and why not merge it into soak?

**Comment by rpjohnst** [comment]
Agreed, there should be more docs - what is there is on the trait, but it's a bit bare-bones.

(Dioptre is its own crate because it is useful on its own for purposes other than `RawTable`, for general field projection, though it could be re-exported as suggested by #2. The name comes from [a unit of optical power](https://en.wikipedia.org/wiki/Dioptre) for [lenses](https://wiki.haskell.org/Lens). 😁)
